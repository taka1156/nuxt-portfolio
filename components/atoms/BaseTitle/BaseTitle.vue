<template>
  <div>
    <h2 class="base-title">
      <!-- @slot タイトル -->
      <slot />
    </h2>
  </div>
</template>

<style scoped>
.base-title {
  font-size: 1.3em;
  line-height: 0em;
  text-align: center;
}
</style>
