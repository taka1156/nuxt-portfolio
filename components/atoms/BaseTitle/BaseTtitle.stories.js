import BaseTitle from './BaseTitle.vue';

export default {
  title: 'Atoms/BaseTitle',
  component: BaseTitle,
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseTitle },
  template: '<base-title>Text</base-title>',
});

export const Default = Template.bind({});
