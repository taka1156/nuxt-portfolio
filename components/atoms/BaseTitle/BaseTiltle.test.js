import { shallowMount } from '@vue/test-utils';
import BaseTitle from './BaseTitle.vue';

describe('BaseTitle', () => {
  it('値がDOMに反映されているか', () => {
    const dummyTitle = 'ダミータイトル';
    const baseTitle = shallowMount(BaseTitle, {
      slots: {
        default: dummyTitle,
      },
    });
    const hTag = baseTitle.find('h2');
    expect(hTag.text()).toBe(dummyTitle);
    expect(baseTitle.html()).toMatchSnapshot();
  });
});
