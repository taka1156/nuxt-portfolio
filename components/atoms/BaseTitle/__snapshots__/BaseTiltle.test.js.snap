// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTitle 値がDOMに反映されているか 1`] = `
"<div>
  <h2 class=\\"base-title\\">ダミータイトル</h2>
</div>"
`;
